### PR TITLE
Enemy AI State Machine

### PR DESCRIPTION
**Pull request by AlexanderGregorSanchez**

### CODE REVIEW

File: Assets/Scripts/Enemy Folder/Enemy.cs [hunk 1]
```diff
@@ -57,74 +80,102 @@ void Update()
...
+        // Attack.Play()
+    }
+
+    public void StopAttackAnimation()
+    {
+        // Attack.Stop()
+    }
+
+
 }
```
Comment (KoolieAid) [inline Assets/Scripts/Enemy Folder/Enemy.cs]: maybe combine this into one function and have a dictionary

add a `throw new NotImplementedException()` to that function

File: Assets/Scripts/Enemy Folder/IdleState.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+using System.Collections.Generic;
+using UnityEngine;
+using Asyncoroutine;
+using UnityEngine.AI;
+using System.Threading;
+
+public class IdleState : MonsterState
+{
+    private float animationDuration;
+    private float animationTimer;
```
Comment (KoolieAid) [inline Assets/Scripts/Enemy Folder/IdleState.cs:9]: multiple classes have the same timer and stuff, why not add an interface

Comment (KoolieAid) [reply Assets/Scripts/Enemy Folder/IdleState.cs:9]: or put on base class

File: Assets/Scripts/Managers & Handlers/MonsterStateManager.cs
Comment (KoolieAid) [inline Assets/Scripts/Managers & Handlers/MonsterStateManager.cs:1]: not bad

**Review by KoolieAid — CHANGES_REQUESTED** [review]

**Comment by KoolieAid** [comment]
ill uhh try to resolve the conflict introduced by #5

File: Assets/Scripts/Enemy Folder/Enemy.cs (hunk 1 above)
Comment (AlexanderGregorSanchez) [reply Assets/Scripts/Enemy Folder/Enemy.cs]: something like this?

`

     public void ControlAnimations(string animationName, bool isPlaying)
     {
        SerializedDictionary<string, Animation> clips = enemyDataInstance.Animations;
        Animation aniClip;
        clips.TryGetValue(animationName, out aniClip);

        if (aniClip == null) return;

        if (isPlaying)
        {
            aniClip.Play();
        }
        else
        {
            aniClip.Stop();
        }
        throw new NotImplementedException();
}

`

File: Assets/Scripts/Enemy Folder/IdleState.cs (hunk 2 above)
Comment (AlexanderGregorSanchez) [reply Assets/Scripts/Enemy Folder/IdleState.cs:9]: at the moment these are just stand-ins. just need to finalize the stats/components/etc of the enemy to take there place

File: Assets/Scripts/Enemy Folder/Enemy.cs (hunk 1 above)
Comment (KoolieAid) [reply Assets/Scripts/Enemy Folder/Enemy.cs]: how about
```csharp
public void PlayAnimation(string name) {
    dict[name].Play();
}

public void StopAnimation(string name) {
    dict[name].Stop();
}
```

add whatever's needed for your `dict` to work this is just some pseudo code

File: Assets/Scripts/Enemy Folder/Enemy.cs [hunk 3]
```diff
@@ -58,73 +89,85 @@ void Update()
...
+
+        if (isPlaying)
+        {
+            aniClip.Play();
+        }
+        else
+        {
+            aniClip.Stop();
+        }
+    }
```
Comment (KoolieAid) [inline Assets/Scripts/Enemy Folder/Enemy.cs]: how about
```csharp
public void PlayAnimation(string index) {
    enemyDataInstance.Animations[index]?.Play();
}

public void Stop Animations(string index) {
    enemyDataInstance.Animations[index]?.Stop();
}
```

**Review by KoolieAid — CHANGES_REQUESTED** [review]

File: Assets/Scripts/Enemy Folder/Enemy.cs (hunk 3 above)
Comment (AlexanderGregorSanchez) [reply Assets/Scripts/Enemy Folder/Enemy.cs]: like this?

public void ControlAnimations(MonsterStates animationName, bool isPlaying)
    {
        if (isPlaying)
        {
            enemyDataInstance.Animations[animationName]?.Play();
        }
        else
        {
            enemyDataInstance.Animations[animationName]?.Stop();
        }
    }

**Review by KoolieAid — APPROVED** [review]

**Pull request merged**